tests/unit/query_v2_create_expand_multiframe.cpp

```diff
@@ -0,0 +1,89 @@
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <memory>
+#include "mock_helpers.hpp"
+
+#include "query/v2/bindings/frame.hpp"
+#include "query/v2/bindings/symbol_table.hpp"
+#include "query/v2/common.hpp"
+#include "query/v2/context.hpp"
+#include "query/v2/plan/operator.hpp"
+#include "query/v2/requests.hpp"
+#include "storage/v3/property_value.hpp"
+#include "storage/v3/shard.hpp"
+#include "utils/logging.hpp"
+#include "utils/memory.hpp"
+
+namespace memgraph::query::v2 {
+
+MultiFrame CreateMultiFrame(const size_t max_pos, const Symbol &src, const Symbol &dst, MockedRequestRouter *router) {
+  static constexpr size_t frame_size = 100;
+  MultiFrame multi_frame(max_pos, frame_size, utils::NewDeleteResource());
+  auto frames_populator = multi_frame.GetInvalidFramesPopulator();
+  size_t i = 0;
+  for (auto &frame : frames_populator) {
+    frame.MakeValid();
+    auto &src_acc = frame.at(src);
+    auto &dst_acc = frame.at(dst);
+    auto v1 = msgs::Vertex{.id = {{msgs::LabelId::FromUint(1)}, {msgs::Value(static_cast<int64_t>(i++))}}};
+    auto v2 = msgs::Vertex{.id = {{msgs::LabelId::FromUint(1)}, {msgs::Value(static_cast<int64_t>(i++))}}};
+    std::map<msgs::PropertyId, msgs::Value> mp;
+    src_acc = TypedValue(query::v2::accessors::VertexAccessor(v1, mp, router));
+    dst_acc = TypedValue(query::v2::accessors::VertexAccessor(v2, mp, router));
+  }
+
+  return multi_frame;
+}
+
+TEST(CreateExpandTest, Cursor) {
+  using testing::_;
+  using testing::Return;
+
+  AstStorage ast;
+  SymbolTable symbol_table;
+
+  plan::NodeCreationInfo node;
+  plan::EdgeCreationInfo edge;
+  edge.edge_type = msgs::EdgeTypeId::FromUint(1);
+  edge.direction = EdgeAtom::Direction::IN;
+  edge.symbol = symbol_table.CreateSymbol("e", true);
+  auto id_alloc = IdAllocator(0, 100);
+
+  const auto &src = symbol_table.CreateSymbol("n", true);
+  node.symbol = symbol_table.CreateSymbol("u", true);
+
+  auto once_cur = plan::MakeUniqueCursorPtr<MockedCursor>(utils::NewDeleteResource());
+  EXPECT_CALL(BaseToMock(*once_cur), PullMultiple(_, _)).Times(1);
+
+  std::shared_ptr<plan::LogicalOperator> once_op = std::make_shared<MockedLogicalOperator>();
+  EXPECT_CALL(BaseToMock(*once_op), MakeCursor(_)).Times(1).WillOnce(Return(std::move(once_cur)));
+
+  auto create_expand = plan::CreateExpand(node, edge, once_op, src, true);
+  auto cursor = create_expand.MakeCursor(utils::NewDeleteResource());
+
+  MockedRequestRouter router;
+  EXPECT_CALL(router, CreateExpand(_))
+      .Times(1)
+      .WillOnce(Return(std::vector<msgs::CreateExpandResponse>{msgs::CreateExpandResponse{}}));
+  auto context = MakeContext(ast, symbol_table, &router, &id_alloc);
+  auto multi_frame = CreateMultiFrame(context.symbol_table.max_position(), src, node.symbol, &router);
```

Comment:
I don't like that the `MultiFrame` is not populated in the `MockedCursor`. This way the cursor might be completely wrong and still managing to satisfy this test.

Reply:
Now we are using the real Once

Reply:
I will leave this as is for now. It is enough to check specific cases in case of somebody needs to debug something, it checks some very basic things and the more complex tests will be done on higher level.